dev/com.ibm.ws.security.mp.jwt_fat/fat/src/com/ibm/ws/security/mp/jwt/fat/sharedTests/MPJwtMPConfigTests.java

```diff
@@ -0,0 +1,628 @@
+            deployRSServerMPConfigInAppInMetaInfApp(server, MpJwtFatConstants.GOOD_JWKSURI_KEYLOCATION_IN_CONFIG_IN_META_INF_ROOT_CONTEXT,
+                                                    buildFileContent(notSet, fixedJwksUri, mpConfigSettings.issuer));
+            deployRSServerMPConfigInAppUnderWebInfApp(server, MpJwtFatConstants.GOOD_RELATIVE_KEYLOCATION_IN_CONFIG_UNDER_WEB_INF_ROOT_CONTEXT,
+                                                      buildFileContent(notSet, PemFile, mpConfigSettings.issuer));
+            deployRSServerMPConfigInAppUnderWebInfApp(server, MpJwtFatConstants.GOOD_FILE_KEYLOCATION_IN_CONFIG_UNDER_WEB_INF_ROOT_CONTEXT,
+                                                      buildFileContent(notSet, fileLoc + PemFile, mpConfigSettings.issuer));
+            deployRSServerMPConfigInAppUnderWebInfApp(server, MpJwtFatConstants.GOOD_FILE_COMPLEX_KEYLOCATION_IN_CONFIG_UNDER_WEB_INF_ROOT_CONTEXT,
+                                                      buildFileContent(notSet, fileLoc + ComplexPemFile, mpConfigSettings.issuer));
+            deployRSServerMPConfigInAppUnderWebInfApp(server, MpJwtFatConstants.GOOD_URL_KEYLOCATION_IN_CONFIG_UNDER_WEB_INF_ROOT_CONTEXT,
+                                                      buildFileContent(notSet, "file:///" + fileLoc + PemFile, mpConfigSettings.issuer));
+            deployRSServerMPConfigInAppUnderWebInfApp(server, MpJwtFatConstants.GOOD_JWKSURI_KEYLOCATION_IN_CONFIG_UNDER_WEB_INF_ROOT_CONTEXT,
+                                                      buildFileContent(notSet, fixedJwksUri, mpConfigSettings.issuer));
+
+            deployRSServerMPConfigInAppInMetaInfApp(server, MpJwtFatConstants.BAD_FILE_KEYLOCATION_IN_CONFIG_IN_META_INF_ROOT_CONTEXT,
+                                                    buildFileContent(notSet, fileLoc + BadPemFile, mpConfigSettings.issuer));
+            deployRSServerMPConfigInAppInMetaInfApp(server, MpJwtFatConstants.BAD_URL_KEYLOCATION_IN_CONFIG_IN_META_INF_ROOT_CONTEXT,
+                                                    buildFileContent(notSet, "file:///" + fileLoc + "someKey.pem", mpConfigSettings.issuer));
+            deployRSServerMPConfigInAppUnderWebInfApp(server, MpJwtFatConstants.BAD_RELATIVE_KEYLOCATION_IN_CONFIG_UNDER_WEB_INF_ROOT_CONTEXT,
+                                                      buildFileContent(notSet, "badPublicKeyLocation", mpConfigSettings.issuer));
+
+        } catch (Exception e) {
+            Log.info(thisClass, "MPJwtAltConfig", "Hit an exception updating the war file" + e.getMessage());
+            throw e;
+        }
+
+    }
+
+    private static void deployRSServerNoMPConfigInAppApp(LibertyServer server) throws Exception {
+        List<String> classList = new ArrayList<String>();
+        classList.add("com.ibm.ws.jaxrs.fat.microProfileApp.CommonMicroProfileMarker_MPConfigNotInApp");
```

Comment:
I'm not sure if the classes are visible here at build time, but would it make sense to do something like `CommonMicroProfileMarker_MPConfigNotInApp.class.getName()` instead of hard coding the full package and class name in a string? Just thinking if we need to move a class or rename a package, we would know something was changed at build time rather than run time.
```suggestion
        classList.add(CommonMicroProfileMarker_MPConfigNotInApp.class.getName());
```

Reply:
If the classes are visible at build time, it could be done for some classes.  For some, they same class exists under multiple apps, so, just specifying the class would be ambiguous.